ppml/README.md

```diff
@@ -0,0 +1,505 @@
+sc.write(df2, AES_CBC_PKCS5PADDING)
+.mode("overwrite")
+.option("header", "true")
+.csv(encryptedOutputPath)
+```
+
+**write mode**
+
+there are 5 modes:
+
+- `overwrite`: Overwrite existing data with the content of dataframe.
+- `append`: Append content of the dataframe to existing data or table.
+- `ignore`: Ignore current write operation if data / table already exists without any error.
+- `error`: Throw an exception if data or table already exists.
+- `errorifexists`: Throw an exception if data or table already exists.
+
+##### PARQUET
+
+Code Example
+
+```scala
+import com.intel.analytics.bigdl.ppml.PPMLContext
+import com.intel.analytics.bigdl.ppml.crypto.{AES_GCM_CTR_V1, PLAIN_TEXT}
+
+// read a plain parquet file and return a DataFrame
+val plainParquetPath = "/plain/parquet/path"
+val df1 = sc.read(PLAIN_TEXT).parquet(plainParquetPath)
+
+// write a DataFrame as a plain parquet file
+plainOutputPath = "/plain/output/path"
+sc.write(df1, PLAIN_TEXT)
+.mode("overwrite")
+.parquet(plainOutputPath)
+
+// read a encrypted parquet file and return a DataFrame
+val encryptedParquetPath = "/encrypted/parquet/path"
+val df2 = sc.read(AES_GCM_CTR_V1).parquet(encryptedParquetPath)
+
+// write a DataFrame as a encrypted parquet file
+val encryptedOutputPath = "/encrypted/output/path"
+sc.write(df2, AES_GCM_CTR_V1)
+.mode("overwrite")
+.parquet(encryptedOutputPath)
+```
+
+##### JSON
+
+Code Example
+
+```scala
+import com.intel.analytics.bigdl.ppml.PPMLContext
+import com.intel.analytics.bigdl.ppml.crypto.{AES_CBC_PKCS5PADDING, PLAIN_TEXT}
+
+// read a plain json file and return a DataFrame
+val plainJsonPath = "/plain/json/path"
+val df1 = sc.read(PLAIN_TEXT).json(plainJsonPath)
+
+// write a DataFrame as a plain json file
+val plainOutputPath = "/plain/output/path"
+sc.write(df1, PLAIN_TEXT)
+.mode("overwrite")
+.json(plainOutputPath)
+
+// read a encrypted json file and return a DataFrame
+val encryptedJsonPath = "/encrypted/parquet/path"
+val df2 = sc.read(AES_CBC_PKCS5PADDING).json(encryptedJsonPath)
+
+// write a DataFrame as a encrypted parquet file
+val encryptedOutputPath = "/encrypted/output/path"
+sc.write(df2, AES_CBC_PKCS5PADDING)
+.mode("overwrite")
+.json(encryptedOutputPath)
+```
+
+##### textfile
+
+Code Example
+
+```scala
+import com.intel.analytics.bigdl.ppml.PPMLContext
+import com.intel.analytics.bigdl.ppml.crypto.{AES_CBC_PKCS5PADDING, PLAIN_TEXT}
+
+// read from a plain csv file and return a RDD
+val plainCsvPath = "/plain/csv/path"
+val rdd1 = sc.textfile(plainCsvPath) // the default cryptoMode is PLAIN_TEXT
+
+// read from a encrypted csv file and return a RDD
+val encryptedCsvPath = "/encrypted/csv/path"
+val rdd2 = sc.textfile(path=encryptedCsvPath, cryptoMode=AES_CBC_PKCS5PADDING)
+```
+
+### 4.2 PPMLContext In Python
+
+We also provide `PPMLContext` Python API. Please refer to [this](https://github.com/intel-analytics/BigDL/blob/main/python/ppml/src/bigdl/ppml/README.md) 
```

Comment:
To use `PPMLContext` Python API, please refer to [PPMLContext Python API](https://github.com/intel-analytics/BigDL/blob/main/python/ppml/src/bigdl/ppml/README.md).